app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -244,255 +246,323 @@ export default class TransactionStateManager extends EventEmitter {
     const network = this.getNetwork();
     const chainId = this.getCurrentChainId();
 
-    // Filter out the ones from the current account and network
-    const otherAccountTxs = txs.filter(
-      (txMeta) =>
-        !(
-          txMeta.txParams.from === address &&
-          transactionMatchesNetwork(txMeta, chainId, network)
-        ),
-    );
-
     // Update state
-    this._saveTxList(otherAccountTxs);
+    this.store.updateState({
+      transactions: omitBy(
+        transactions,
+        (transaction) =>
+          transaction.txParams.from === address &&
+          transactionMatchesNetwork(transaction, chainId, network),
+      ),
+    });
+  }
+
+  /**
+   * Filters out the unapproved transactions from state
+   */
+  clearUnapprovedTxs() {
+    this.store.updateState({
+      transactions: omitBy(
+        this.store.getState().transactions,
+        (transaction) => transaction.status === TRANSACTION_STATUSES.UNAPPROVED,
+      ),
+    });
   }
 
   //
   //           PRIVATE METHODS
   //
 
   /**
-   * @param {number} txId - the txMeta Id
-   * @param {TransactionStatuses[keyof TransactionStatuses]} status - the status to set on the txMeta
-   * @emits tx:status-update - passes txId and status
-   * @emits ${txMeta.id}:finished - if it is a finished state. Passes the txMeta
-   * @emits 'updateBadge'
+   * Updates a transaction's status in state, and then emits events that are
+   * subscribed to elsewhere. See below for best guesses on where and how these
+   * events are received.
+   * @param {number} txId - the TransactionMeta Id
+   * @param {TransactionStatusString} status - the status to set on the
+   *  TransactionMeta
+   * @emits txMeta.id:txMeta.status - every time a transaction's status changes
+   *  we emit the change passing along the id. This does not appear to be used
+   *  outside of this file, which only listens to this to unsubscribe listeners
+   *  of :rejected and :signed statuses when the inverse status changes. Likely
+   *  safe to drop.
+   * @emits tx:status-update - every time a transaction's status changes we
+   *  emit this event and pass txId and status. This event is subscribed to in
+   *  the TransactionController and re-broadcast by the TransactionController.
+   *  It is used internally within the TransactionController to try and update
+   *  pending transactions on each new block (from blockTracker). It's also
+   *  subscribed to in metamask-controller to display a browser notification on
+   *  confirmed or failed transactions.
+   * @emits txMeta.id:finished - When a transaction moves to a finished state
+   *  this event is emitted, which is used in the TransactionController to pass
+   *  along details of the transaction to the dapp that suggested them. This
+   *  pattern is replicated across all of the message managers and can likely
+   *  be supplemented or replaced by the ApprovalController.
+   * @emits 'updateBadge' - When the number of transactions changes in state,
```

Comment:
Nit: It would be nice to standardize our syntax for JSDoc events - either quoted or not.

Reply:
i went without quotes, per the jsdocs documentation for @fires